adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java

```diff
@@ -101,14 +126,34 @@ public void deleteAllDatasources() {
       throws DatasourceNotFoundException, ImporterParameterException, InterpreterParameterException, IOException {
     Datasource datasource = getDatasource(id);
     AdapterConfig adapterConfig = datasource.toAdapterConfig(runtimeParameters);
-    DataImportResponse executionResult = adapter.executeJob(adapterConfig);
-    DataImport dataImport = new DataImport(datasource, executionResult.getData());
+    try {
+      DataImportResponse executionResult = adapter.executeJob(adapterConfig);
+
+      DataImport dataImport = new DataImport(datasource, executionResult.getData());
 
-    DataImport savedDataImport = dataImportRepository.save(dataImport);
+      DataImport savedDataImport = dataImportRepository.save(dataImport);
 
-    amqpPublisher.publishImportSuccess(id, savedDataImport.getData());
+      amqpPublisher.publishImportSuccess(id, savedDataImport.getData());
 
-    return savedDataImport.getMetaData();
+      if (datasource.getSchema() != null) {
```

Comment:
Has to happen before DataImport is created

Reply:
As i need a dataImport in order to call validate according to the other review, this line of code is transfared right after the dataImport creation but before the DB save